src/com/palmergames/bukkit/towny/event/DeletePlayerEvent.java

```diff
@@ -18,15 +17,28 @@ public static HandlerList getHandlerList() {
         return handlers;
     }
 
-    public DeletePlayerEvent(String player) {
-        super(!Bukkit.getServer().isPrimaryThread());
-        this.playerName = player;
+    public DeletePlayerEvent(Resident resident) {
+        super(resident.getName(), resident.getUUID(), resident.getRegistered());
     }
 
     /**
      * @return the deleted player's name.
      */
     public String getPlayerName() {
-        return playerName;
+        return name;
     }
+
+	/**
+	 * @return deleted player's uuid
+	 */
+	public UUID getPlayerUUID() {
```

Comment:
This one here could be a bit of trouble. It would return null if the resident has not had a UUID applied to it, (which could certainly happen if they haven't logged in since UUID collection began and the old-resident-delete is why they're deleted,) or if the admin has the UUID gathering task turned off. Maybe put a warning that it could return null in the javadoc?